src/esp/geo/geo.cpp

```diff
@@ -39,13 +39,12 @@ std::vector<vec2f> convexHull2D(const std::vector<vec2f>& points) {
   }
 
   // Build upper hull
-  for (size_t i = idx.size() - 2, t = k + 1; i >= 0; i--) {
```

Comment:
And here, just to have a history record for all this deep changes.

Reply:
Resolves a segfault that occurs on Linux due to undefined behavior of unsigned integer underflows.